base/event.jl

```diff
@@ -2,6 +2,39 @@
 
 ## condition variables
 
+
+if JULIA_PARTR
+
+import Core.Condition
+
+Condition() = ccall(:jl_condition_new, Ref{Condition}, ())
+wait(c::Condition) = ccall(:jl_task_wait, Cvoid, (Ref{Condition},), Ref{c})
+notify(c::Condition, arg, all, error) = ccall(:jl_task_notify, Cvoid, (Ref{Condition},), Ref{c})
+notify(c::Condition, @nospecialize(arg = nothing); all=true, error=false) = notify(c, arg, all, error)
+notify_error(c::Condition, err) = notify(c, err, true, true)
+n_waiters(c::Condition) = 0 # TODO: do we need this?
+
+yield() = ccall(:jl_task_yield, Cvoid, (Cint,), 1)
+wait() = yield()
```

Comment:
`wait()` should block forever (until the task is scheduled again). `yield()` should switch away from the current task, but immediately put it back in a queue so it will run again.

Reply:
Yes, you could call `schedule` on it, or it might just block forever and get GC'd.